src/scripts/main.js

```diff
@@ -1,24 +1,39 @@
+/* eslint-disable no-shadow */
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    element.addEventListener(eventName, function handler(event) {
+      element.removeEventListener(eventName, handler);
+
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}`,
+      );
+    });
+  });
 }
 
 const printMessage = (message) => {
-  // write your code here
+  const div = document.createElement('div');
+
+  div.classList.add('message');
+  div.textContent = message;
+  document.body.appendChild(div);
 };
 
-const loginField = document.getElementById('login');
-const passwordField = document.getElementById('password');
-const button = document.getElementById('submit');
+document.addEventListener('DOMContentLoaded', () => {
+  const loginField = document.getElementById('login');
+  const passwordField = document.getElementById('password');
+  const button = document.getElementById('submit');
 
-waitFor(loginField, 'click').then(printMessage);
-waitFor(passwordField, 'click').then(printMessage);
-waitFor(button, 'click').then(printMessage);
+  waitFor(loginField, 'click').then(printMessage);
+  waitFor(passwordField, 'click').then(printMessage);
+  waitFor(button, 'click').then(printMessage);
```

Comment:
Consider whether you need to listen for 'click' events on the login and password fields. Typically, 'click' events are more relevant for buttons or interactive elements. If the intention is to capture user interaction, ensure this aligns with your requirements.